state/relationunit.go

```diff
@@ -317,12 +317,14 @@ func (ru *RelationUnit) LeaveScope() error {
 			Remove: true,
 		}}
 		if ru.relation.doc.Life == Alive {
-			ops = append(ops, txn.Op{
-				C:      relationsC,
-				Id:     ru.relation.doc.DocID,
-				Assert: bson.D{{"life", Alive}},
-				Update: bson.D{{"$inc", bson.D{{"unitcount", -1}}}},
-			})
+			if ru.checkUnitLife {
```

Comment:
the txn.Op below isn't just checking life, it's also decrementing unitcount. I don't think you want to stop decrementing...

Reply:
We do because the same code prevents the increment in enter scope. Without this code, the unitcount gets out of sync. For ru's belonging to the remote app side, we don't want to monitor unit count at all.

Reply:
please rename checkUnitLife as discussed